common/lib/xmodule/xmodule/license.py

```diff
@@ -0,0 +1,138 @@
+"""
+Different types of licenses for content
+
+This file contains a base license class as well as a some useful specific license classes, namely:
+    - ARRLicense (All Rights Reserved License)
+    - CCLicense (Creative Commons License)
+
+The classes provide utility funcions for dealing with licensing, such as getting an image representation
+of a license, a url to a page describing the specifics of the license, converting licenses to and from json
+and storing some vital information about licenses, in particular the version.
+"""
+from django.utils.translation import ugettext as _
+
+from xblock.fields import JSONField
+
+
+class License(JSONField):
+    """
+    Base License class
+    """
+
+    _default = None
+    MUTABLE = False
+
+    def __init__(self, kind=None, version=None, *args, **kwargs):
+        self.kind = kind
+        self.version = version
+        super(License, self).__init__(*args, **kwargs)
+
+    @property
+    def html(self):
+        """
+        Return a piece of html that describes the license
+
+        This method should be overridden in child classes to provide the desired html.
+        """
+        return u"<p>" + _("This resource is not licensed.") + u"</p>"
```

Comment:
@explorerleslie q: what should the very base, default license readout be? Nominally, this should _never_ be displayed, because all license types should override the html message with their own message. However, I feel like this is perhaps a dangerous default/worst-case-scenario, and perhaps we should consider going with "All Rights Reserved"?